### ISSUE TITLE
Viewer node causes constant "Scene Changed" updates

### ISSUE THREAD

**Issue by gerroon** [JacquesLucke/animation_nodes#1860]
The view node causes scene changed evaluation constantly.

Load the attached .blend file. Then open the Nodetree panel and see what is happening. Delete the Viewer node to observe that the scene changed updates stop.

Unpluging it does not stop this behavior, it seems to evaluate constantly.

2.3.0 Windows 11, tested with Blender 3.1 and 3.3

[_05062022_1325_15.zip](https://github.com/JacquesLucke/animation_nodes/files/8840968/_05062022_1325_15.zip)

**Comment by OmarEmaraDev** [comment]
I can replicate that. Not immediately sure what is causing it. I will take a look.

**Comment by Durman** [comment]
I guess that Sverchok has similar problem after adding update trigger upon depsgraph changes. Evaluation of a tree can cause changes in a scene what can call depsgraph trigger again and the execution falls into infinite loop. In Sverchok I'm adding skip_next_update key to the tree what seems works unless depsgraph trigger is not called several times in a row.

**Comment by OmarEmaraDev** [comment]
@Durman Thanks for the hint. Though, AN doesn't store any runtime data in IDs in most cases, so we probably don't need something as elaborate as what Sverchok has. We just need to handle it on a case by case basis.

---

It seems the Viewer node changes the node tree during execution, causing a depsgraph tag that is somehow missed in the two pass evaluation of the depsgraph.

```
graph_id_tag_update: id=NTNodeTree flags=COPY_ON_WRITE source=USER_EDIT
```

This is likely because the viewer node draws some of its content on the socket itself by changing its name, so it essentially changes the node tree. In the `setViewData` method:

```
self.inputs[0].name = nodeText
```

So perhaps the solution is to just avoid drawing something in the socket and always use UI extensions. Same applies for node width adjustments. This needs to be investigated from a UI point of view first.

**Comment by OmarEmaraDev** [comment]
I will look into it this weekend. Do you think always drawing the data inside a UI Extension even for single values is reasonable? UI Extensions are the white boxs drawn under nodes like in the case where the viewer node is viewing a list.

**Comment by OmarEmaraDev** [comment]
Fixed in dd5241f. Thanks for the report!
